tests/php/unit/Controller/BookingControllerTest.php

```diff
@@ -151,23 +167,108 @@ public function testGetBookableSlotsInvalidTimezone(): void {
+		;
+
 		$apptConfg = new AppointmentConfig();
 		$apptConfg->setId(1);
 		$this->time->expects(self::once())
 			->method('getTime')
-			->willReturn($fakeFutureTimestamp);
-		$this->apptService->expects(self::never())
+			->willReturn($now);
+		$this->apptService->expects(self::once())
 			->method('findById')
-			->with(1);
-		$this->bookingService->expects(self::never())
-			->method('getAvailableSlots');
-		$this->logger->expects(self::once())
-			->method('warning');
+			->with(1)
+			->willReturn($apptConfg);
+		$this->bookingService->expects(self::once())
+			->method('getAvailableSlots')
+			->with($apptConfg, $sDT, $eDT);
 
-		$this->controller->getBookableSlots($apptConfg->getId(), $start, 'Europe/Berlin');
+		$this->controller->getBookableSlots($apptConfg->getId(), $start, $timezone);
+	}
+
+	public function testGetBookableSlotsTimezoneMinus10(): void {
+
+		$now = (new DateTime('2024-6-30 8:00:00'))->getTimestamp();
+		$start = (new DateTime('2024-7-1 4:00:00'))->getTimestamp(); // Start date with America/Toronto offset
+		$timezone = 'Pacific/Pago_Pago';
+		$sDT = (new DateTime('2024-7-1 11:00:00'))->getTimestamp();
+		;
+		$eDT = (new DateTime('2024-7-2 11:00:00'))->getTimestamp();
+		;
+
+		$apptConfg = new AppointmentConfig();
+		$apptConfg->setId(1);
+		$this->time->expects(self::once())
+			->method('getTime')
+			->willReturn($now);
+		$this->apptService->expects(self::once())
+			->method('findById')
+			->with(1)
+			->willReturn($apptConfg);
+		$this->bookingService->expects(self::once())
+			->method('getAvailableSlots')
+			->with($apptConfg, $sDT, $eDT);
+
+		$this->controller->getBookableSlots($apptConfg->getId(), $start, $timezone);
+	}
+
+	public function testGetBookableSlotsTimezonePlus10(): void {
+
+		$now = (new DateTime('2024-6-30 8:00:00'))->getTimestamp();
+		$start = (new DateTime('2024-7-1 4:00:00'))->getTimestamp(); // Start date with America/Toronto offset
+		$timezone = 'Australia/Sydney';
+		$sDT = (new DateTime('2024-6-30 14:00:00'))->getTimestamp();
+		;
+		$eDT = (new DateTime('2024-7-1 14:00:00'))->getTimestamp();
+		;
+
+		$apptConfg = new AppointmentConfig();
+		$apptConfg->setId(1);
+		$this->time->expects(self::once())
+			->method('getTime')
+			->willReturn($now);
+		$this->apptService->expects(self::once())
+			->method('findById')
+			->with(1)
+			->willReturn($apptConfg);
+		$this->bookingService->expects(self::once())
+			->method('getAvailableSlots')
+			->with($apptConfg, $sDT, $eDT);
+
+		$this->controller->getBookableSlots($apptConfg->getId(), $start, $timezone);
+	}
+
+	public function testGetBookableSlotsTimezonePlus14(): void {
+
+		$now = (new DateTime('2024-6-30 8:00:00'))->getTimestamp();
+		$start = (new DateTime('2024-7-1 4:00:00'))->getTimestamp(); // Start date with America/Toronto offset
+		$timezone = 'Pacific/Kiritimati';
+		$sDT = (new DateTime('2024-6-30 10:00:00'))->getTimestamp();
+		;
+		$eDT = (new DateTime('2024-7-1 10:00:00'))->getTimestamp();
+		;
```

Comment:
```suggestion
```

Reply:
Yeah not sure how those got in there. Corrected.